Add target age group filter and fix place selector in Events app search

nikomakela opened the pull request:
TH-1327.

1. Target Age Group -filter added in the advanced search form of Events app. There are keywords mapped by the age group, that should be used when searching the events from LinkedEvents. The group for adults is different, because it negates the other groups' keywords.
2. Added a FilterSummary feature for the new filter.
3. Fixed the place filter selector.
4. Removed Only children events -checkbox filter.
5. Use shallow rendering on search.
6. Fix search form related styles and make them consistent.

<img width="1340" height="748" alt="image" src="https://github.com/user-attachments/assets/905251c5-0f9e-4195-8196-b1fc138b9796" />

<img width="1026" height="910" alt="image" src="https://github.com/user-attachments/assets/4b2f4899-a867-463b-a7e5-f537af8aab25" />

<img width="1257" height="787" alt="image" src="https://github.com/user-attachments/assets/2120e325-97ce-4937-a25c-11f6767e0715" />


<img width="734" height="846" alt="image" src="https://github.com/user-attachments/assets/279e5af2-15ce-456a-9ee2-1ecf050d9611" />

<img width="483" height="905" alt="image" src="https://github.com/user-attachments/assets/2b313796-496d-4ef3-a864-5cd9afaeea94" />

nikomakela committed:
feat(events): add target group filter to search form

TH-1327.

Target Age Group -filter added in the advanced search form of Events
app.

There are keywords mapped by the age group, that should be used when
searching the events from LinkedEvents. The group for adults is
different, because it negates the other groups' keywords.

Also added a FilterSummary feature for the new filter.

nikomakela committed:
feat(events): remove unused "place" -filter handler

TH-1327.

The event search have had a place URL-parameter handler, which has been
a string (a single id), that should have been mapped with a constant
value, but since the constant was an empty object, the feature must not
have worked. Because of that, the place-filter ("place") is now removed.
The "places" filter still exists and is also read from URL.

nikomakela committed:
fix(events): place selector and multi select dropdown

TH-1327.

Cloned the missing implementation from Kultus UI, to fix selected optons
rendering issues in PlaceSelector and to add the FixedOptions feature to
MultiSelectDropdown.

Ref 1. https://github.com/City-of-Helsinki/palvelutarjotin-ui/
blob/bd4a0447fbdc3d5b9d5ed18f20f03f2d1e3c885c/
src/common/components/multiSelectDropdown/MultiSelectDropdown.tsx

REf 2. https://github.com/City-of-Helsinki/palvelutarjotin-ui/
blob/bd4a0447fbdc3d5b9d5ed18f20f03f2d1e3c885c
/src/domain/place/placeSelector/PlaceSelector.tsx

nikomakela committed:
feat(events): use shallow rendering when searching events

TH-1327.

Use shallow rendering when making a new search, so the page does not
double render, make the UX laggy and the search slow. This feature was
already implemented in useGoToSearch -hook, but the hook was not in use.

Shallow routing allows you to change the URL without running data
fetching methods again, that includes getServerSideProps,
getStaticProps, and getInitialProps.
See more:
https://nextjs.org/docs/pages/building-your-application/routing/linking-and-navigating

nikomakela committed:
style: remove unused no-explicit-any

TH-1327

nikomakela committed:
fix(events): consistent styles for filter elements

TH-1327.

Fix the font sizes and svg-icon sizes of the Events-Helsinki search form
filters.

nikomakela committed:
feat(events): remove only children events filter

TH-1327.

nikomakela committed:
fix(events): apply theme to target age group selector

TH-1327.

Apply "green app theme" to new target age group selector, since the
original highlight color is blue and a greenish one is needed.

WIP

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr828.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr828.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr828.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr828.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr828.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr828.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr828.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr828.dev.hel.ninja :crying_cat_face::anger::boom::boom:

karisal-anders commented on apps/events-helsinki/src/domain/search/eventSearch/advancedSearch/TargetAgeGroupSelector.tsx:
```diff
@@ -0,0 +1,96 @@
+  return targetGroups;
+}
+
+export type TargetAgeGroupOptionType = ReturnType<
+  typeof useTargetAgeGroupSelectorOptions
+>[number];
```
Using something like this type here could bring more clarity/transparency to what this type is. And a bit more strictness as the current type does not specify that the value needs to be AgeGroup or empty string, but the return type is (probably, not 100% sure) resolved as `{label: string; value: string}`:
```typescript
type TargetAgeGroupOptionType = { label: string; value: AgeGroup | '' };
```
Not a big deal, but just was a bit curious to see what the type was and thought about to what specificity can Typescript see the return type's values (i.e. the use of map function is the one that makes the "as const" mapping lose information for the target group values AFAIK).

karisal-anders replied:
It seems that other places don't accept the stricter definition of `value` field here.
So maybe just simplifying to:
```typescript
export type TargetAgeGroupOptionType = { label: string; value: string };
```

karisal-anders replied:
I'd go with the simplest solution here for clarity, i.e. the OptionType and I'd give the function that should return it it's return type as OptionType[]. That's very simple, enforces that the function does return such a type, and clear for the user to see what the type is. But, I'm not going to force this issue here, I am giving my own recommendation, and you can do as you see is best here.

karisal-anders commented on apps/events-helsinki/src/constants.ts:
```diff
@@ -51,3 +51,65 @@ export const FEEDBACK_LINKS = {
+ * @see https://helsinkisolutionoffice.atlassian.net/browse/TH-1327 for spec.
+ */
+export const AGE_GROUP_KEYWORDS: Record<AgeGroup, readonly string[]> = {
+  babies: ['yso:p20513', 'yso:p15937'], // Vauvaperheet
```
Small thing: could give description of both yso keywords:
```suggestion
  babies: [
    'yso:p20513', // Vauvaperheet
    'yso:p15937', // Vauvat
  ], 
```

karisal-anders reviewed (approved):
Left some comments, please take a look at them.

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr828.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr828.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr828.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
refactor(events): add adult keyword exclusion for "playgrounds"

TH-1327.

Product owner asked "playgrounds" keyword to be added to adults
exclusion in target age group filter. Also, the `kulke:355` was not
needed.

nikomakela committed:
fix(events): disable autocomplete of search input

TH-1327.

Disable the autocomplete from search input. NOTE: this  does not affect
to autocomplete menu, which is another side-effect. This is only for the
input's autocomplete.

nikomakela committed:
refactor(events): change order of target age group and date filters

TH-1327

Fix style issues in date selector and target age group filters.

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr828.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr828.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr828.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
fix(events): style of empty option in target age group selector

TH-1327.

The empty option was shown with really narrow height when any value was
given for target age group selector. This style adds a minimum height,
even for an empty content.

nikomakela committed:
fix(events): accessibility issues in target age group selector

TH-1327.

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr828.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr828.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr828.dev.hel.ninja :laughing::tada::tada::tada:
